core/src/main/java/com/linecorp/armeria/common/CoreBlockhoundIntegration.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
+
+import java.util.ResourceBundle;
+import java.util.UUID;
+
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+import reactor.blockhound.BlockHound.Builder;
+import reactor.blockhound.integration.BlockHoundIntegration;
+
+/**
+ * A {@link BlockHoundIntegration} for the core module.
+ */
+@UnstableApi
+public final class CoreBlockhoundIntegration implements BlockHoundIntegration {
+    @Override
+    public void applyTo(Builder builder) {
+        // short locks
+        builder.allowBlockingCallsInside("com.linecorp.armeria.client.HttpClientFactory",
+                                         "pool");
```

Comment:
Could you explain which part of the method is a blocking call? 🤔

Reply:
It seems like the report from @mauhiz contains the stacktrace. I was able to reproduce this failure from our CI as well.

https://github.com/line/armeria/pull/4493#issuecomment-1403088000
```
at java.base/java.util.concurrent.locks.ReentrantLock.lock(ReentrantLock.java:267)
	at com.linecorp.armeria.internal.shaded.guava.collect.MapMakerInternalMap$Segment.put(MapMakerInternalMap.java:1476)
	at com.linecorp.armeria.internal.shaded.guava.collect.MapMakerInternalMap.putIfAbsent(MapMakerInternalMap.java:2419)
	at java.base/java.util.concurrent.ConcurrentMap.computeIfAbsent(ConcurrentMap.java:331)
	at com.linecorp.armeria.client.HttpClientFactory.pool(HttpClientFactory.java:404)
	at com.linecorp.armeria.client.HttpClientDelegate.acquireConnectionAndExecute0(HttpClientDelegate.java:169)
```

Reply:
Got it. Thanks! 😉